File: threat-detection-platform/backend/app/schemas/incident.py
"""
Incident Schemas
"""

from typing import Dict, List, Optional

from pydantic import BaseModel, Field


class IncidentCreateRequest(BaseModel):
    """Manual incident creation (by operators)."""

    camera_id: str = Field(description="Source camera ID")
    threat_type: str = Field(description="Threat type (handgun, rifle, knife)")
    description: str = Field(default="", max_length=1000, description="Incident description")
    risk_level: str = Field(default="MEDIUM", description="Risk level override")


class IncidentStatusUpdate(BaseModel):
    """Update incident status."""

    status: str = Field(description="New status: ACKNOWLEDGED, RESOLVED, FALSE_POSITIVE")
    notes: Optional[str] = Field(default=None, max_length=2000, description="Resolution notes")


class IncidentResponse(BaseModel):
    """Incident detail response."""

    id: str
    incident_number: int
    camera_id: str
    threat_type: str
    confidence: float
    risk_score: float
    risk_level: str
    status: str
    track_id: int | None = None
    location_id: str | None = None
    location_name: str = ""
    building: str = ""
    zone: str = ""
    latitude: float | None = None
    longitude: float | None = None
    weapon_count: int = 1
    frames_confirmed: int = 0
    description: str = ""
    acknowledged_by: str | None = None
    acknowledged_at: str | None = None
    resolved_by: str | None = None
    resolved_at: str | None = None
    resolution_notes: str | None = None
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class IncidentListResponse(BaseModel):
    """Incident list item (summary)."""

    id: str
    incident_number: int
    camera_id: str
    threat_type: str
    risk_score: float
    risk_level: str
    status: str
    location_name: str = ""
    latitude: float | None = None
    longitude: float | None = None
    created_at: str

    model_config = {"from_attributes": True}

File: threat-detection-platform/frontend/src/types/incident.ts
import { IncidentStatus, RiskLevel } from './common';

export interface Incident {
  id: string;
  incident_number: number;
  camera_id: string;
  threat_type: string;
  confidence: number;
  risk_score: number;
  risk_level: RiskLevel;
  status: IncidentStatus;
  track_id: number | null;
  location_id: string | null;
  location_name: string;
  building: string;
  zone: string;
  latitude: number | null;
  longitude: number | null;
  weapon_count: number;
  frames_confirmed: number;
  description: string;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  resolved_by: string | null;
  resolved_at: string | null;
  resolution_notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface IncidentListItem {
  id: string;
  incident_number: number;
  camera_id: string;
  threat_type: string;
  risk_score: number;
  risk_level: RiskLevel;
  status: IncidentStatus;
  location_name: string;
  latitude: number | null;
  longitude: number | null;
  created_at: string;
}
